### PR TITLE
Orange

### PR DESCRIPTION
**Pull request by zzmp**

### CODE REVIEW

File: libraries/gpu/src/gpu/GLBackendInput.cpp [hunk 1]
```diff
@@ -264,16 +315,25 @@ void GLBackend::resetInputStage() {
     glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, 0);
     (void) CHECK_GL_ERROR();
 
+#if defined(SUPPORT_VERTEX_ATTRIB_FORMAT)
+    glBindBuffer(GL_ARRAY_BUFFER, 0);
+    for (uint32_t i = 0; i < _input._attributeActivation.size(); i++) {
+        glDisableVertexAttribArray(i);
+    }
+    for (uint32_t i = 0; i < _input._attribBindingBuffers.size(); i++) {
```
Comment (zzmp) [inline libraries/gpu/src/gpu/GLBackendInput.cpp]: This is only set in updatInput

**Comment by hifi-gustavo** [comment]
Build console output for commit: f5e7a113a79d82572444db3943bf3836cde4e2eb available at: https://gist.github.com/183acc2dcd54ca5cfc467771849b7748<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7920.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7920.dmg

File: libraries/gpu/src/gpu/GLBackend.h [hunk 2]
```diff
@@ -397,12 +407,14 @@ class GLBackend : public Backend {
...
         bool _invalidFormat = true;
         Stream::FormatPointer _format;
+        std::string _formatKey;
 
         typedef std::bitset<MAX_NUM_ATTRIBUTES> ActivationCache;
         ActivationCache _attributeActivation;
 
         typedef std::bitset<MAX_NUM_INPUT_BUFFERS> BuffersState;
         BuffersState _invalidBuffers;
+        BuffersState _attribBindingBuffers;
```
Comment (zzmp) [inline libraries/gpu/src/gpu/GLBackend.h]: Does this need to be initialized?

File: libraries/gpu/src/gpu/GLBackendInput.cpp [hunk 3]
```diff
@@ -12,12 +12,42 @@
...
+}
+
 void GLBackend::do_setInputFormat(Batch& batch, size_t paramOffset) {
     Stream::FormatPointer format = batch._streamFormats.get(batch._params[paramOffset]._uint);
-
-    if (format != _input._format) {
-        _input._format = format;
-        _input._invalidFormat = true;
+    GLInputFormat* ifo = GLBackend::syncGPUObject(*format);
+    if (ifo) {
```
Comment (zzmp) [inline libraries/gpu/src/gpu/GLBackendInput.cpp]: This is a new check

File: libraries/gpu/src/gpu/GLBackendInput.cpp [hunk 4]
```diff
@@ -12,12 +12,42 @@
...
+        if (format != _input._format) {
+            _input._format = format;
+            if (format) {
+                if (_input._formatKey != format->getKey()) {
+                    _input._formatKey = format->getKey();
+                    _input._invalidFormat = true;
```
Comment (zzmp) [inline libraries/gpu/src/gpu/GLBackendInput.cpp]: This is duplicated

**Comment by zzmp** [comment]
Opened accidentally, meant to be local.

**Pull request closed without merging**